docs/DATA.md

```diff
@@ -0,0 +1,88 @@
+
+```
+Note: This system is probably important enough that we give it its own name, that is not just purely descriptive. 
+```
+
+Why have a single interface for so many different types of datastores? After all, expected data both in and out varies greatly between the different types of datastores. However, there are several major advantages of a single data access layer across all of Habitat:
+1. Interoperable applications are way easier to build if they all use the same API for accessing data.
+2. Consolidating the access point for these datastores reduces the number of ports that have to be opened on each node, simplifying the system.
+3. Common proxying logic can be implemented to forward requests for data not immediately available on the queried node's machine. 
+4. A common authorization and permissions system can be implemented as middleware for this data access API. This also increases the interoperability of applications.
+
+## The API
+
+The data layer is served over `http`. Using the Habitat proxy, HTTP requests can be made over a `LibP2P` transport layer to bypass NATs. The URL for accessing data on any datastore follows this pattern:
+
+`https://<node_host>:<reverse_proxy_port>/data/<community_id>/<datastore>/<namespace>/<content_identifer>`
+
+The components:
+* `node_host`: IP or domain name for the node being queried
+* `reverse_proxy_port`: The node's reverse proxy port, default 2041
+* `community_id`: The id for the community storing the data you are trying to access
+* `datastore`: The datastore that the request should be routed towards
+* `namespace`: The namespace that the data is under.
+* `content_identifier`: String the datastore uses to find the data you are querying.
+
+Any HTTP method can be used, as long as the datastore the request is routed to supports it. The `content_identifier` format will differ between datastores, and can be anything from a file path to a content hash, to a SQL query.
+
+## Namespaces
+Namespaces allows for data to be grouped across datastores. This is useful for permissions and authorization, as well as simply giving a high level grouping for data that can be utilized by applications. The namespace has analogues in the different types of datastores. In a RDBMS, a namespace corresponds roughly to a database, while in a filesystem a namespace might correspond to a disk partition or drive. The most common use of namespaces is to group data written by the same application to different datastrores. Every datasource must partition data by namespace.
+
+Namespace conflicts can arise if two different applications choose to write to the same namespace. `TODO we need to figure out a way to resolve this`.
```

Comment:
shouldn't applications be able to read/write the same namespaces? i thought this is a feature